java/search-for-webmap/src/main/java/com/esri/arcgisruntime/sample/searchforwebmap/MainActivity.java

```diff
@@ -0,0 +1,231 @@
+        PortalItemAdapter portalItemAdapter = new PortalItemAdapter(mPortalItemList,
+            portalItem -> addMap(mPortal, portalItem.getItemId()));
+        mRecyclerView.setAdapter(portalItemAdapter);
+        // open the drawer once there are results
+        mDrawer.openDrawer(Gravity.START);
+        // get 10 more results to fill the recycler view
+        getMoreResults();
+      } catch (Exception e) {
+        String error = "Error getting portal query result set: " + e;
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+        Log.e(TAG, error);
+      }
+    });
+  }
+
+  /**
+   * Adds the next set of results to the recycler view.
+   */
+  private void getMoreResults() {
+    if (mPortalQueryResultSet.getNextQueryParameters() != null) {
+      // find the next 10 matching portal items
+      mMoreResults = mPortal.findItemsAsync(mPortalQueryResultSet.getNextQueryParameters());
+      mMoreResults.addDoneListener(() -> {
+        try {
+          // replace the result set with the current set of results
+          mPortalQueryResultSet = mMoreResults.get();
+          List<PortalItem> portalItems = mPortalQueryResultSet.getResults();
+          // add results to the recycler view
+          mPortalItemList.addAll(portalItems);
+          mRecyclerView.getAdapter().notifyDataSetChanged();
+          mRecyclerView.smoothScrollToPosition(mRecyclerView.getAdapter().getItemCount() - 9);
+        } catch (Exception e) {
+          String error = "Error getting portal query result set: " + e;
+          Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+          Log.e(TAG, error);
+        }
+      });
+    } else {
+      Toast.makeText(this, "There are no more results matching this query", Toast.LENGTH_LONG).show();
+    }
+  }
+
+  /**
+   * Add the given portal item to a new map and set the map to the map view.
+   *
+   * @param portal
+   * @param itemId
+   */
+  private void addMap(Portal portal, String itemId) {
+    // report error and return if portal is null
+    if (portal == null) {
+      String error = "Portal not instantiated.";
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+      Log.e(TAG, error);
+      return;
+    }
+    // use the item ID to create a portal item from the portal
+    PortalItem portalItem = new PortalItem(portal, itemId);
+    // create a map using the web map (portal item) and add it to the map view
+    ArcGISMap webMap = new ArcGISMap(portalItem);
+    mMapView.setMap(webMap);
+    // close the drawer
+    mDrawer.closeDrawer(Gravity.START);
+    // check if webmap is supported
+    mMapView.getMap().addDoneLoadingListener(() -> {
+      if (mMapView.getMap().getLoadError() != null) {
+        String error = "Unable to load map: " + mMapView.getMap().getLoadError().getMessage();
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+        Log.e(TAG, error);
+      }
+    });
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
```

Comment:
I somehow don't see a menu in the app, the left arrow on top left of the app, opens up to an empty sidebar